nerfstudio/models/kplanes.py

```diff
@@ -0,0 +1,396 @@
+        {"img": 1.0, "plane_tv": 0.01, "plane_tv_proposal_net": 0.0001, "distortion": 0.001, "interlevel": 1.0}
+    )
+
+
+class KPlanesModel(Model):
+    """K-Planes model
+
+    Args:
+        config: K-Planes configuration to instantiate model
+    """
+
+    config: KPlanesModelConfig
+
+    def populate_modules(self):
+        """Set the fields and modules."""
+        super().populate_modules()
+
+        if self.config.is_contracted:
+            scene_contraction = SceneContraction(order=float("inf"))
+        else:
+            scene_contraction = None
+
+        # Fields
+        self.field = KPlanesField(
+            self.scene_box.aabb,
+            num_images=self.num_train_data,
+            grid_config=self.config.grid_config,
+            concat_features_across_scales=self.config.concat_features_across_scales,
+            multiscale_res=self.config.multiscale_res,
+            spatial_distortion=scene_contraction,
+            appearance_embedding_dim=self.config.appearance_embedding_dim,
+            use_average_appearance_embedding=self.config.use_average_appearance_embedding,
+            linear_decoder=self.config.linear_decoder,
+            linear_decoder_layers=self.config.linear_decoder_layers,
+        )
+
+        self.density_fns = []
+        num_prop_nets = self.config.num_proposal_iterations
+        # Build the proposal network(s)
+        self.proposal_networks = torch.nn.ModuleList()
+        if self.config.use_same_proposal_network:
+            assert len(self.config.proposal_net_args_list) == 1, "Only one proposal network is allowed."
+            prop_net_args = self.config.proposal_net_args_list[0]
+            network = KPlanesDensityField(
+                self.scene_box.aabb,
+                spatial_distortion=scene_contraction,
+                linear_decoder=self.config.linear_decoder,
+                **prop_net_args,
+            )
+            self.proposal_networks.append(network)
+            self.density_fns.extend([network.density_fn for _ in range(num_prop_nets)])
+        else:
+            for i in range(num_prop_nets):
+                prop_net_args = self.config.proposal_net_args_list[min(i, len(self.config.proposal_net_args_list) - 1)]
+                network = KPlanesDensityField(
+                    self.scene_box.aabb,
+                    spatial_distortion=scene_contraction,
+                    linear_decoder=self.config.linear_decoder,
+                    **prop_net_args,
+                )
+                self.proposal_networks.append(network)
+            self.density_fns.extend([network.density_fn for network in self.proposal_networks])
+
+        # Samplers
+        update_schedule = lambda step: np.clip(
+            np.interp(step, [0, self.config.proposal_warmup], [0, self.config.proposal_update_every]),
+            1,
+            self.config.proposal_update_every,
+        )
+
+        if self.config.is_contracted or self.config.is_ndc:
```

Comment:
Here there's no point in using the `is_ndc` config as nerfstudio doesn't have ndc datasets